File: solutions/2026-04-15/python/2026-04-15-shortest-distance-to-target-string-in-a-circular-array.py
```python
# Summary: Find the shortest distance to a target string in a circular array, starting from a given index.
# Link: https://leetcode.com/problems/shortest-distance-to-target-string-in-a-circular-array/
# Approach:
# 1. Iterate through the `words` array to find all indices where `target` appears.
# 2. For each found index, calculate the distance from `startIndex` considering the circular nature of the array.
#    The distance can be calculated in two ways:
#    a. Moving clockwise: (found_index - startIndex + n) % n
#    b. Moving counter-clockwise: (startIndex - found_index + n) % n
#    The shortest distance for a specific found index is the minimum of these two.
# 3. Keep track of the overall minimum distance found.
# 4. If `target` is not found in the array, return -1. Otherwise, return the overall minimum distance.
# Time Complexity: O(n), where n is the length of `words`. We iterate through the array once to find occurrences of `target` and calculate distances.
# Space Complexity: O(1), as we only use a few variables to store the minimum distance and loop indices.

def shortestDistance(words, target, startIndex):
    """
    Calculates the shortest distance to reach the target string in a circular array.

    Args:
        words: A list of strings representing the circular array.
        target: The string to find.
        startIndex: The starting index in the array.

    Returns:
        The shortest distance to reach the target string, or -1 if not found.
    """
    n = len(words)
    min_dist = float('inf')  # Initialize minimum distance to infinity
    found = False  # Flag to check if target is found

    # Iterate through the array to find occurrences of the target string
    for i in range(n):
        if words[i] == target:
            found = True  # Mark that target has been found
            # Calculate clockwise distance
            dist_clockwise = (i - startIndex + n) % n
            # Calculate counter-clockwise distance
            dist_counter_clockwise = (startIndex - i + n) % n
            # The shortest distance for this occurrence is the minimum of clockwise and counter-clockwise
            current_min_dist = min(dist_clockwise, dist_counter_clockwise)
            # Update the overall minimum distance
            min_dist = min(min_dist, current_min_dist)

    # If target was not found, return -1. Otherwise, return the calculated minimum distance.
    if not found:
        return -1
    else:
        return min_dist

# Example Usage:
# Example 1:
# words1 = ["hello", "i", "am", "leetcode", "hello"]
# target1 = "hello"
# startIndex1 = 1
# print(f"Example 1 Output: {shortestDistance(words1, target1, startIndex1)}")  # Expected: 1

# Example 2:
# words2 = ["a", "b", "leetcode"]
# target2 = "leetcode"
# startIndex2 = 0
# print(f"Example 2 Output: {shortestDistance(words2, target2, startIndex2)}")  # Expected: 1

# Example 3:
# words3 = ["i", "eat", "leetcode"]
# target3 = "ate"
# startIndex3 = 0
# print(f"Example 3 Output: {shortestDistance(words3, target3, startIndex3)}")  # Expected: -1
```

File: solutions/2026-04-15/java/2026-04-15-shortest-distance-to-target-string-in-a-circular-array.java
```java
// Problem: Shortest Distance to Target String in a Circular Array
// Link: https://leetcode.com/problems/shortest-distance-to-target-string-in-a-circular-array/
// Approach:
// The problem asks for the shortest distance to a target string in a circular array, starting from a given index.
// Since the array is circular, we can move left or right. For each occurrence of the target string, we calculate the distance from the startIndex.
// The distance can be calculated in two ways: moving right or moving left.
// For a target at index `i`:
// - Distance moving right: `(i - startIndex + n) % n`
// - Distance moving left: `(startIndex - i + n) % n`
// We need to find the minimum of these two distances for all occurrences of the target.
// If the target is not found, return -1.
//
// Time Complexity: O(n*m), where n is the length of `words` and m is the average length of strings in `words`.
// We iterate through the `words` array once (O(n)). Inside the loop, we compare strings.
// In the worst case, string comparison can take O(m) time.
//
// Space Complexity: O(1) (excluding input storage). We only use a few variables to store the minimum distance.
class Solution {
    public int shortestDistance(String[] words, String target, int startIndex) {
        int n = words.length; // Get the length of the words array
        int minDistance = Integer.MAX_VALUE; // Initialize minimum distance to a very large value
        boolean found = false; // Flag to check if the target string is found

        // Iterate through each word in the array
        for (int i = 0; i < n; i++) {
            // Check if the current word matches the target string
            if (words[i].equals(target)) {
                found = true; // Mark that the target has been found

                // Calculate the distance by moving right (clockwise)
                // (i - startIndex + n) % n handles wrap-around for circular array
                int distanceRight = (i - startIndex + n) % n;

                // Calculate the distance by moving left (counter-clockwise)
                // (startIndex - i + n) % n handles wrap-around for circular array
                int distanceLeft = (startIndex - i + n) % n;

                // Update the minimum distance with the shorter of the two calculated distances
                minDistance = Math.min(minDistance, Math.min(distanceRight, distanceLeft));
            }
        }

        // If the target string was not found, return -1
        if (!found) {
            return -1;
        }

        // Return the shortest distance found
        return minDistance;
    }
}
```